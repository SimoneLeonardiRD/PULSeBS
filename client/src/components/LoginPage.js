import React from "react";
import {
  Form,
  FormGroup,
  Col,
  Button,
  FormControl,
  Container,
  Row,
} from "react-bootstrap";
// import { ROLES } from "../data/consts";
import { AuthContext } from "../auth/AuthContext";

class LoginPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      // role: ROLES.TEACHER,
      //   submitted: false,
      //   error: "",
      //   has_error: false,
    };
  }

  handleInputChange = (event) => {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });
  };

  handleSubmit = (event, onLogin) => {
    event.preventDefault();
    onLogin(this.state.username, this.state.password);
  };

  render() {
    return (
      <AuthContext.Consumer>
        {(context) => (
          <Container fluid className="center center mt-5 ">
            <Row>
              <Col className="center center mt-5 " md={{ span: 4, offset: 4 }}>
                <h3 className="center mb-3"> Login </h3>

                <Form
                  method="POST"
                  onSubmit={(event) =>
                    this.handleSubmit(event, context.loginUser)
                  }
                >
                  {/* <FormGroup>
                    <select
                      className="form-control"
                      role={this.state.role}
                      onChange={this.handleInputChange}
                      name="role"
                    >
                      <option value="teacher">Teacher</option>
                      <option value="student">Student</option>
                      <option value="teacher">Booking Manager</option>
                      <option value="student">Support Officer</option>
                    </select>
                  </FormGroup> */}
                  <FormGroup>
                    <FormControl
                      type="text"
                      id="username"
                      name="username"
                      placeholder="Username"
                      value={this.state.username}
                      onChange={this.handleInputChange}
                      required
                      autoFocus
                    />
                  </FormGroup>
                  <FormGroup>
                    <FormControl
                      type="password"
                      id="password"
                      name="password"
                      placeholder="Password"
                      value={this.state.password}
                      onChange={this.handleInputChange}
                      required
                    />
                  </FormGroup>

                  <FormGroup>
                    <Button
                      block
                      variant="primary"
                      type="submit"
                      color="primary"
                      size="sm"
                    >
                      Login
                    </Button>
                  </FormGroup>
                </Form>
              </Col>
            </Row>
          </Container>
        )}
      </AuthContext.Consumer>
    );
  }
}

export default LoginPage;
