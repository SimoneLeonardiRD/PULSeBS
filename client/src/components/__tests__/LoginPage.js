import React from "react";
import LoginPage from "../../components/LoginPage";
import renderer from "react-test-renderer";

test("Fist SnapShot test ", () => {
  const tree = renderer.create(<LoginPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
