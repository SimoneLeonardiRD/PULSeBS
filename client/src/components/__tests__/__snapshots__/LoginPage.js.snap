// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fist SnapShot test  1`] = `
<div
  className="center center mt-5 container-fluid"
>
  <div
    className="row"
  >
    <div
      className="col-md-4 offset-md-4"
    >
      <h3>
         Select your Role: 
      </h3>
      <form
        className=""
        method="POST"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <select
            className="form-control"
            name="role"
            onChange={[Function]}
            role="teacher"
          >
            <option
              value="teacher"
            >
              Teacher
            </option>
            <option
              value="student"
            >
              Student
            </option>
            <option
              value="teacher"
            >
              Booking Manager
            </option>
            <option
              value="student"
            >
              Support Officer
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <input
            autoFocus={true}
            className="form-control"
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="btn btn-primary"
            color="primary"
            disabled={false}
            type="submit"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
